### PR TITLE
US 4051 // Filter dataexplorer rows by type

### PR DESCRIPTION
**Pull request by Dinika**
Fixes #4051
Screenshot from 2023-07-13 12-46-18

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have added necessary unit and integration tests.
- [X] I have added screenshots (if applicable), in the comment section.

### CODE REVIEW

File: src/subapps/dataExplorer/ProjectSelector.tsx [hunk 1]
```diff
@@ -1,47 +1,60 @@
...
   const allOptions = [
     { value: AllProjects, label: AllProjects },
-    ...(projects?.map(projectToOption) ?? []),
-  ].filter(project => project.value.includes(searchTerm));
+    ...(aggregations?.projects.buckets.map(projectToOption) ?? []),
+  ];
 
   return (
     <div className="form-container">
       <span className="label">Show me </span>
```
Comment (bilalesi) [inline src/subapps/dataExplorer/ProjectSelector.tsx:24]: the label of the selector: `Project`

File: src/subapps/dataExplorer/TypeSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      optionsRef.current = [
+        ...(aggregations?.types.buckets.map(typeToOption) ?? []),
+      ];
+      setDisplayedOptions(optionsRef.current);
+    }
+  }, [isSuccess, aggregations]);
+
+  return (
+    <div className="form-container">
+      <span className="label">Show me </span>
```
Comment (bilalesi) [inline src/subapps/dataExplorer/TypeSelector.tsx:38]: the label of the selector: `Type`

File: src/subapps/dataExplorer/DataExplorerUtils.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    queryFn: async () => {
+      const allResults = await nexus.Resource.list(
+        orgAndProject?.[0],
+        orgAndProject?.[1],
+        {
+          aggregations: true,
+        }
+      );
+      return ((allResults as unknown) as AggregationsResult).aggregations;
+    },
```
[the remark this answers is not in the document]
Comment (Dinika) [reply src/subapps/dataExplorer/DataExplorerUtils.tsx:19]: That's amazing! Thanks for sharing that :)

File: src/subapps/dataExplorer/ProjectSelector.tsx (hunk 1 above)
Comment (Dinika) [reply src/subapps/dataExplorer/ProjectSelector.tsx:24]: My gosh :sweat_smile: thanks for catching that.

**Review by bilalesi — APPROVED** [review]

**Pull request merged**